addons/website/models/ir_ui_view.py

```diff
@@ -13,6 +13,40 @@
 
 
 class view(osv.osv):
+
+    def _in_menu(self, cr, uid, ids, name, arg, context=None):
+        result = {}
+        website_menu = self.pool.get('website.menu')
+        for page in self.browse(cr, uid, ids, context=context):
+            result[page.id] = False
+            #No website_id means that the page is shared by several websites
+            domain = [('url', '=', "/page/" + page.key.split('.')[-1])]
+            if page.website_id.id:
+                result[page.id] = bool(website_menu.search(cr, uid, domain + [('website_id', '=', page.website_id.id)], limit=1, context=context))
+            else:
+                menu_ids = website_menu.search(cr, uid, domain, context=context)
+                website_ids = [menu.website_id.id for menu in website_menu.browse(cr, uid, menu_ids, context=context)]
+                #To check if a menu points to a shared page
+                for website_id in website_ids:
+                    if not bool(self.search(cr, uid, [('key', '=', page.key), ('website_id', '=', website_id)], limit=1, context=context)):
```

Comment:
no for !
result[page.id] = bool(self.search(cr, uid, [('key', '=', page.key), ('website_id', '=', website_ids)], limit=1, context=context))